react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { isClicked: false };
+  }
+
+  handleClick() {
+    this.setState({ isClicked: true });
+    this.props.text = 'Thank you!';
+  }
+
+  render() {
+    return <button onClick={this.handleClick}>{this.props.text}</button>;
+  }
```

Comment:
Here you should use an `if` statement to set the correct text in the button based on the value in `this.state`, not `this.props`.
```JSX
if (this.state.isClicked) {
  return <button>Thank You!</button>;
} else {
  return <button onClick={this.handleClick}>Click Me!</button>;
}
```